### PR TITLE
Fix SIGFPEs when logging csched progress

### PR DESCRIPTION
**Pull request by tristan957**
Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: lib/cn/csched_sp3.c [hunk 1]
```diff
@@ -409,15 +409,23 @@ sp3_log_progress(struct cn_compaction_work *w, struct cn_merge_stats *ms, bool f
```
Comment (nabeelmmd) [inline lib/cn/csched_sp3.c:409]: Seems like we could clamp est->cwe_keys and avoid the if...else?

```
est->cw_keys = clamp_t(type, est->cw_keys, 1, est->cw_keys);
progress = 1.0 * ms->ms_keys_in / est->cwe_keys;
```

**Review by nabeelmmd — APPROVED** [review]

File: lib/cn/csched_sp3.c (hunk 1 above)
Comment (tristan957) [reply lib/cn/csched_sp3.c:409]: Seems like a fine solution. I will do that.

Comment (tristan957) [reply lib/cn/csched_sp3.c:409]: Implemented, but chose not to change est->cw_keys

Comment (beckerg) [reply lib/cn/csched_sp3.c:409]: Why not just cast to double?

Comment (tristan957) [reply lib/cn/csched_sp3.c:409]: with `1.0` stuff? I don't know. Didn't write the code. Want me to change it?

Comment (tristan957) [reply lib/cn/csched_sp3.c:409]: Changed in latest push

**Review by beckerg — APPROVED** [review]

File: lib/cn/csched_sp3.c [hunk 2]
```diff
@@ -409,15 +409,17 @@ sp3_log_progress(struct cn_compaction_work *w, struct cn_merge_stats *ms, bool f
 
     if (final) {
         msg_type = "final";
-        progress = 1.0 * ms->ms_keys_in / est->cwe_keys;
+        progress = (double)w->cw_stats.ms_keys_in /
+            clamp_t(uint64_t, est->cwe_keys, 1, est->cwe_keys);
```
Comment (alexttx) [inline lib/cn/csched_sp3.c:413]: and yall made fun of my safe_div macro :)

**Pull request merged**